Release process

agarciamontoro opened the pull request:
- [x] Refactor release process
- [x] Release version 9.0 with integrity checks
- [x] Release version 10.0 with integrity checks
- [x] Minimal tests for version existence

I'm leaving the tests for the code for a future PR. I want to release this asap to have some coherence between `db.rnacentral`, `db.rna16s` and `db.fragments16s`.

agarciamontoro committed:
Update dependencies and move data values outside package.scala

agarciamontoro committed:
Move release-related code to main package

agarciamontoro committed:
Refactor release-related code

agarciamontoro committed:
Refactor release-related code

agarciamontoro committed:
Make code more readable

agarciamontoro committed:
Document release code

agarciamontoro committed:
Document data values

agarciamontoro committed:
Parametrize output file with version and local directory

agarciamontoro committed:
Fix broken links in docs

agarciamontoro committed:
Add old version 9.0

agarciamontoro committed:
Bump rnacentral version

agarciamontoro committed:
Remove useless arguments

agarciamontoro committed:
Bump db.rnacentral version

agarciamontoro committed:
Add tests to check existence of supported versions

agarciamontoro committed:
Fix format

agarciamontoro committed:
Update readme

ncordon commented on deps.sbt:
```diff
@@ -1,11 +1,12 @@
 val mainDependencies = Seq(
-  "ohnosequences" %% "aws-scala-tools" % "0.20.0"
+  "ohnosequences" %% "db-rnacentral" % "0.13.1",
+  "ohnosequences" %% "s3"            % "0.2.1",
```
This line can be removed. Already included in `db-rnacentral`

ncordon commented on deps.sbt:
```diff
@@ -1,11 +1,12 @@
-  "ohnosequences" %% "db-rnacentral" % "0.12.2",
-  "ohnosequences" %% "fastarious"    % "0.12.0",
-  "org.scalatest" %% "scalatest"     % "3.0.4"
+  "ohnosequences" %% "fastarious" % "0.12.0",
```
Just a future note: `fastarious` should be removed in the future and changed to `faster`

agarciamontoro replied:
Issue opened: https://github.com/ohnosequences/db.rna16s/issues/95

ncordon commented on deps.sbt:
```diff
@@ -1,11 +1,12 @@
-  "ohnosequences" %% "aws-scala-tools" % "0.20.0"
+  "ohnosequences" %% "db-rnacentral" % "0.13.1",
+  "ohnosequences" %% "s3"            % "0.2.1",
+  "ohnosequences" %% "files"         % "0.5.0"
```
Same

ncordon committed:
Updates dependencies

ncordon commented on src/main/scala/data.scala:
```diff
@@ -0,0 +1,78 @@
+    * obtain.
+    *
+    * @return a function `String => S3Object` that, given the name of a file,
+    * return an S3 object in a fixed S3 directory, which is parametrized by the
```
Typo:
```suggestion
    * returns an S3 object in a fixed S3 directory, which is parametrized by the
```

ncordon commented on src/main/scala/data.scala:
```diff
@@ -0,0 +1,78 @@
+import ohnosequences.files.File
+
+sealed abstract class Version(val name: String) {
+  val compatibleInputVersions: Set[RNACentralVersion]
```
What does it mean to be a compatible version?

agarciamontoro replied:
I just tried to explain why this was there and found no real reason, so I'm removing it. Thanks for making me think a little bit more about this :3

ncordon commented on src/main/scala/release.scala:
```diff
@@ -0,0 +1,116 @@
+    * [[generateNewDB]] for that use case.
+    *
+    * @return an Error + S3Object, with a Right(s3Obj) with the S3 path of the
+    * generated fastaa if everything worked as expected or with a Left(error)
```
```suggestion
    * generated fasta if everything worked as expected or with a Left(error)
```

agarciamontoro committed:
Add notes for release v3.0.0

ncordon commented on src/main/scala/release.scala:
```diff
@@ -0,0 +1,116 @@
+      _ <- getCheckedFileIfDifferent(inputFasta, fastaFile)
+      _ <- getCheckedFileIfDifferent(inputIdMapping, mappingsFile)
+      _ <- generateSequences(rnaCentralEntries, outputFile)
+      _ <- paranoidPutFile(outputFile, s3Sequences)
```
:+1: Nice

agarciamontoro committed:
Rename compatibleInputVersions -> otherCompatibleInputVersions

The input version is, by definition, a compatible version, so it
makes no sense to always add it to the compatibleInputVersions set.
The rename makes clear that this value stores only other compatible
input versions.

ncordon commented on src/main/scala/release.scala:
```diff
@@ -0,0 +1,116 @@
+
+  /**
+    * Read the data from `db.rnacentral`, filter the 16S sequences and upload
+    * the result to S3 if and only if the upload does not override anything.
```
Typo: override -> overwrite
```suggestion
    * the result to S3 if and only if the upload does not overwrite anything.
```

agarciamontoro replied:
Oh, thanks, I thought they were interchangeable! This explains it with nice examples: https://english.stackexchange.com/a/88403

ncordon commented on src/main/scala/release.scala:
```diff
@@ -0,0 +1,116 @@
+    * sequences are filtered using [[rna16sIdentification.is16s]]. The result
+    * of the filter is uploaded to the object returned by [[data.sequences]]
+    * applied over the version parameter, if and only if the upload does not
+    * override anything.
```
```suggestion
    * overwrite anything.
```

agarciamontoro committed:
Fix typo in v3.0.0 notes

agarciamontoro committed:
Fix link in v3.0.0 notes

ncordon reviewed (approved):
Great work on this :muscle: !

ncordon committed:
Fix typo in docs

Co-Authored-By: agarciamontoro <alejandro.garciamontoro@gmail.com>

ncordon committed:
Fix typo in docs

Co-Authored-By: agarciamontoro <alejandro.garciamontoro@gmail.com>

ncordon committed:
Fix misuse of override instead of overwrite

Co-Authored-By: agarciamontoro <alejandro.garciamontoro@gmail.com>

ncordon committed:
Fix another misuse of override instead of overwrite

Co-Authored-By: agarciamontoro <alejandro.garciamontoro@gmail.com>

agarciamontoro committed:
Remove otherCompatibleInputVersions

agarciamontoro committed:
Merge branch 'release.process' of github.com:ohnosequences/db.rna16s into release.process

agarciamontoro commented:
I'll merge this as soon as it is released, thanks for the review, @ncordon

agarciamontoro committed:
Fix typo in release notes for v3.0.0
